### ISSUE TITLE
TBA management

### ISSUE THREAD

**Issue by AndrewVSutherland** [roed314/seminars#503]
A number of organizers have started filling out their schedules with TBA speakers.  I think it would be useful to have an option to filter out talks where the speaker is TBA on the main browse page (and possibly also where the title is TBA).  I think we should also automatically convert "tba" to "TBA" for consistency.

For talk titles, blank and TBA are equivalent everywhere except on the schedule page.  We could automatically convert TBA titles to blank when saving/loading talks -- this would have no visible impact except possibly on the schedule page (we could fill in TBA there too, but maybe better not to so organizers can just type in the title without deleting the TBA, but maybe we want to highlight schedule slots with TBA titles in some way).

It might also be a good idea to email a reminder to the organizers/curators who are creators whenever there is a talk with the speaker or title still set to TBA within 24 (or maybe 48?) hours of the talk starting, as we are planning to do for missing live links for online talks when the access control is not set to external registration.

**Comment by roed314** [comment]
I will add a TBA filter in the More section.

**Comment by poonen** [comment]
I'm not sure that a TBA filter is a good idea.  The filter collection is already quite complicated, and this one seems not so useful.

**Comment by AndrewVSutherland** [comment]
tba/TBA titles are now converted to blanks when saved, I agree that it probably doesn't make sense to add a TBA filter to the more knowl.  That leaves open only the question of whether we want to nudge organizers to fill in TBA's or not -- as a curator of many seminars, I would actually appreciate getting an email reminder, say, 36 hours before a talk that still has a TBA speaker or title.

**Comment by poonen** [comment]
I was able to save a talk title "TBA" on the Edit schedule page.

Also, I don't understand the intended behavior here.  Are you automatically erasing talk titles entered as "TBA", and then displaying them as "TBA" even though the talk title is now blank?
To me it seems a little heavy-handed to be changing what people enter unless there is a good reason to do so.  Why are we doing this?

**Comment by poonen** [comment]
I don't think we should be sending emails to organizers about TBA titles/speakers, unless they explicitly opt in to receive such emails.

**Comment by AndrewVSutherland** [comment]
Blank and TBA talk titles are indistinguishable to anyone browsing talks on  our site, empty titles are always interpreted as TBA.. It's generally bad practice to store data with the same meaning in two different ways (e.g. if you want to search for talks that have TBA titles or track statistics, you would have to search for the empty string, TBA, tba, and possibly many foreign language variants of TBA). This is why organizers are encouraged to leave TBA titles blank rather than typing in TBA.  Another advantage of using the empty string is that it is language independent (so if you wanted to customize our site to display TBA in a different languages depending on the viewer, if they are all empty strings, this is easier to do than if you have to compute all possible translations -- TBA means the same thing in every language, even though if we don't all write it the same way).

I completely agree with you about opting-in to receive email reminders.

**Comment by poonen** [comment]
OK, yes, I agree that we should encourage users to enter an empty title instead of TBA, and I'm guess you are saying that you are enforcing this by automatically converting "TBA" entered in the Edit talk page to blank.  Presumably in this case, you should do the same if TBA is entered
as a title on the Edit schedule page.

Do you also convert "TBD" and "To be announced" and "tba" titles, etc., to blank?

If we are going to be automatically changing what people enter, this should be said.  The oxymoronic warning saying "TBA title left blank (it will appear as TBA)" is confusing because it doesn't say _where_ it is going to be left blank and _where_ it will appear as TBA.  On the Edit talk page?  On the Browse page?  On the View talk page?  In the talk knowl?  In the series schedule?

What about TBA speakers?

When you say "interpreted as TBA", do you mean that if the user enters nothing in the title box on the Edit talk page, then the talk will appear in the Browse page and in the View talk page and in series schedules with the title "TBA"?

**Comment by AndrewVSutherland** [comment]
Currently it is only converting TBA and tba (which were the main ones people were using) but I can add the variations you suggested.  I agree that this should also be done on the Edit schedule page and that the message displayed should be made clearer, and should probably be just an informational message not a warning.

What I meant be "interpreted as TBA" is exactly what you wrote, and I believe this has been the case since we first launched the site, we never display blank titles.  This issue is about trying to make the existing behavior transparent and more uniform.

We currently do not allow the speaker names to be blank or treat the speaker name "TBA" any differently than any other speaker name.  It might make sense to change this, but we would have to think about how best support this on the edit schedule page.

**Comment by poonen** [comment]
I edited the Title knowl to clarify this.

**Comment by poonen** [comment]
It would be good if you or someone could implement those other changes:
1) conversion of TBD, To be announced, To be determined (regardless of capitalization)
2) same conversion on Edit schedule page
3) change the warning to an informational message (I already edited its text).

**Comment by poonen** [comment]
Let's leave the behavior of TBA speakers as it is, for now.

**Comment by AndrewVSutherland** [comment]
Sounds good, I can address 1-3 tomorrow.

**Comment by AndrewVSutherland** [comment]
Done.  The only other thing I might suggest in the short term is, rather than sending any (opt-in) emails, we could simply highlight series that have a TBA title (or speaker) for a talk happening in the next week (put an asterisk by them on the Manage series page).  I would find this useful for the series I am curating (currently I have to look at all of them every few days).

This could be done in conjunction with highlighting series whose external pages have been updated more recently than the most recent update to a talk in the series on researchseminars.org, see #494.  The goal is the same: draw the organizers attention to series they may want to update.

Other than that I think I'm happy to close this issue.

**Comment by poonen** [comment]
Your new suggestion sounds good, but how would organizers know that the asterisk meant this?

**Comment by AndrewVSutherland** [comment]
I was thinking we could display an informational message at the top of the page saying something like "Starred series have talks with TBA titles within the next week" (only displayed if any are actually starred), and similarly for external website updates.  Alternatively we could add a new table listing "talks that may need attention" that would appear only when non-empty -- this would make it easy to see exactly where you might need to make updates and put you one click away from making them.

**Comment by AndrewVSutherland** [comment]
In fact we could go a bit further and add an "organizer's dashboard" to the manage page that lists recently completed talks (to which you might want to post slides or video recordings, Jamie Sully has suggested nudging organizers to do this) and talks coming up soon, along with a note about any actions that might be required.

**Comment by poonen** [comment]
OK, maybe on the Manage page just above the section "Create a series" add a new section called
"Talks in recent past and near future"
containing a single table listing all talks in increasing chronological order from one week ago to two weeks in the future (so past talks first), with past and future separated by a horizontal bar.
The table would look like

Speaker               Title                                                   Warnings
Bjorn Poonen      Grand discovery       edit  view         

and under Warnings we would see red text like
"no slides, no video" for past talks, and
"speaker TBA, title TBA" for past and future talks.
The table should be present only when there are talks in it.
Maybe the first row under "Speaker..."
should contain ...
and the last should contain ...
to indicate that these are not necessarily all the talks.


As for external websites of series, I think I wouldn't create an additional table for that,
but just put red asterisks at the upper right of the series' names, with an informational message at the top saying something like
"Red asterisks indicate series whose external website has changed".
